core/analytics/src/main/java/com/appcoins/wallet/core/analytics/analytics/legacy/BillingAnalytics.kt

```diff
@@ -217,58 +349,78 @@ class BillingAnalytics @Inject constructor(
-        context,
-        txId,
+        context = context,
+        uid = txId,
         packageName = packageName,
         usdAmount = valueUsd,
         appcAmount = value
       )
     }
 
     eventData[EVENT_OEMID] = oemIdPreferencesDataSource.getCurrentOemId()
-    analytics.logEvent(eventData, WALLET_PAYMENT_CONCLUSION, AnalyticsManager.Action.CLICK, WALLET)
+    analytics.logEvent(
+      eventData,
+      WALLET_PAYMENT_CONCLUSION,
+      AnalyticsManager.Action.CLICK,
+      WALLET
+    )
     appStartPreferencesDataSource.saveIsFirstPayment(isFirstPayment = false)
   }
 
   override fun sendPaymentPendingEvent(
-    packageName: String, skuDetails: String?, value: String,
-    purchaseDetails: String, transactionType: String, isOnboardingPayment: Boolean
+    packageName: String,
+    skuDetails: String?,
+    value: String,
+    purchaseDetails: String,
+    transactionType: String,
+    isOnboardingPayment: Boolean
   ) {
     val eventData: MutableMap<String, Any?> = createConclusionWalletEventMap(
-      packageName, skuDetails, value, purchaseDetails,
-      transactionType, EVENT_PENDING, isOnboardingPayment
+      packageName = packageName,
+      skuDetails = skuDetails,
+      value = value,
+      purchaseDetails = purchaseDetails,
+      transactionType = transactionType,
+      status = EVENT_PENDING,
+      isOnboardingPayment = isOnboardingPayment
+    ).apply { EVENT_OEMID to oemIdPreferencesDataSource.getCurrentOemId() }
+
+    analytics.logEvent(
+      eventData,
+      WALLET_PAYMENT_CONCLUSION,
+      AnalyticsManager.Action.CLICK,
+      WALLET
     )
-    eventData[EVENT_OEMID] = oemIdPreferencesDataSource.getCurrentOemId()
-    analytics.logEvent(eventData, WALLET_PAYMENT_CONCLUSION, AnalyticsManager.Action.CLICK, WALLET)
   }
 
   override fun sendPurchaseStartEvent(
-    packageName: String?, skuDetails: String?, value: String,
-    purchaseDetails: String, transactionType: String?, context: String, isOnboardingPayment: Boolean
+    packageName: String?,
+    skuDetails: String?,
+    value: String,
+    transactionType: String?,
+    context: String,
+    oemId: String?,
+    purchaseDetails: String?,
+    isOnboardingPayment: Boolean
   ) {
-    val eventData: MutableMap<String, Any?> = HashMap()
-    eventData[EVENT_PACKAGE_NAME] = packageName
-    eventData[EVENT_SKU] = skuDetails
-    eventData[EVENT_VALUE] = value
-    eventData[EVENT_TRANSACTION_TYPE] = transactionType
-    eventData[EVENT_PAYMENT_METHOD] = purchaseDetails
-    eventData[EVENT_CONTEXT] = context
-    if (isOnboardingPayment) eventData[EVENT_ONBOARDING_PAYMENT] = true
-    analytics.logEvent(eventData, WALLET_PAYMENT_START, AnalyticsManager.Action.CLICK, WALLET)
-  }
+    val eventData = mutableMapOf<String, Any?>(
+      EVENT_PACKAGE_NAME to packageName,
+      EVENT_SKU to skuDetails,
+      EVENT_VALUE to value,
+      EVENT_TRANSACTION_TYPE to transactionType,
+      EVENT_CONTEXT to context,
+      OEM_ID to oemId,
```

Comment:
oemId field will need to be empty string in case of not having a oemId. This is needed as a way to differentiate between not having a oemId and not supporting oemId in the event